pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/loader/SegmentDirectoryLoaderContext.java

```diff
@@ -28,22 +29,30 @@
 public class SegmentDirectoryLoaderContext {
 
   private final TableConfig _tableConfig;
+  private final Schema _schema;
   private final String _instanceId;
   private final String _segmentName;
   private final PinotConfiguration _segmentDirectoryConfigs;
+  private final String _segmentCrc;
 
-  public SegmentDirectoryLoaderContext(TableConfig tableConfig, String instanceId, String segmentName,
-      PinotConfiguration segmentDirectoryConfigs) {
+  public SegmentDirectoryLoaderContext(TableConfig tableConfig, Schema schema, String instanceId, String segmentName,
```

Comment:
nit: use a Builder so that it'd be easier to extend in future, which would avoid touching those test cases in future.

Reply:
good suggestion! I should've thought of that lol. Changed it.